### PR TITLE
Improve status display for manually-graded questions

### PR DESCRIPTION
**Pull request by nwalters512**
This PR makes three changes as proposed in #10667. All of these apply only for purely manually-graded questions on an Exam.

- Previously, after saving a submission, the "auto-grading available points" column showed "0" and the "auto-grading awarded points" column shows "pending" once a submission is saved. These now both show &mdash;.
- If someone only saves an answer to a purely manual-graded question, there will be a "grade 1 saved answer" button that does absolutely nothing. Now, the button is disabled and says "No saved answers to grade".
- After a submission is saved to the purely manually-graded question, the submission status was shown as a teal "saved" badge. Now, it shows as a green "saved for manual grading" badge.

### CODE REVIEW

**Comment by nwalters512** [comment]
IIRC while testing, I saw weird behavior when a question explicitly had zero points. I need to check that again.

**Comment by jonatanschroeder** [comment]
> If someone only saves an answer to a purely manual-graded question, there will be a "grade 1 saved answer" button that does absolutely nothing. Now, the button is disabled and says "No saved answers to grade".

Does this resolve #3555?

**Comment by nwalters512** [comment]
@jonatanschroeder I believe it will! If you have the bandwidth, I'd appreciate your help testing this once I take it out of draft given your expertise on manual grading.

File: apps/prairielearn/src/components/QuestionScore.html.ts [hunk 1]
```diff
@@ -249,12 +249,33 @@ export function ExamQuestionScore({
...
   };
+  assessment_question: Pick<AssessmentQuestion, 'max_auto_points' | 'max_manual_points'>;
 }) {
+  // Special case: if this is a manually graded question in the "saved" state,
+  // we want to differentiate it from saved auto-graded questions which can
+  // be graded immediately. We'll use a green badge so that student can drive
+  // towards all status badges being green.
+  //
+  // TODO: can we safely look at the assessment question for exams? What about
+  // the guarantee that an Exam-type assessment won't change after it's created?
```
Comment (nwalters512) [inline apps/prairielearn/src/components/QuestionScore.html.ts:266]: 👀 I'd appreciate some help thinking through this case.

File: apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts [hunk 2]
```diff
@@ -45,6 +45,21 @@ export function StudentAssessmentInstance({
     }
   | { groupConfig?: undefined; groupInfo?: undefined; userCanAssignRoles?: undefined }
 )) {
+  let savedAnswers = 0;
+  let suspendedSavedAnswers = 0;
+  resLocals.instance_questions.forEach((question) => {
+    if (question.status === 'saved') {
+      if (question.allow_grade_left_ms > 0) {
+        suspendedSavedAnswers++;
+      } else if (question.max_auto_points) {
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts:54]: This doesn't exactly do the right thing when `"points": [0], "manualPoints": 0` is used. In that case, "Save & grade" is still shown on the instance question page, but that question won't be gradable from the assessment instance page. Is there something better to check here, maybe `question.open && (question.max_auto_points || !question.max_manual_points)`? I think this mirrors the behavior from `lib/question-render.js`.

File: apps/prairielearn/src/components/QuestionScore.html.ts (hunk 1 above)
Comment (mwest1066) [reply apps/prairielearn/src/components/QuestionScore.html.ts:266]: Referring to assessment questions is definitely a problem, but we already do it elsewhere so I don't think we are making things any worse here.

I've been thinking for a long time that we need to improve our support for modifying in-progress assessments (especially exams). I need to write this up, but the key thing is that we should make a per-assessment-instance copy of zones and assessment questions at the time that the assessment instance is created. Ideally we'd also lock question versions once question versioning is implemented. This then allows the assessment instance to be unaffected by any changes to the original assessment. We'd then need to add explicit operations to "reset" an assessment instance to the latest version of the assessment. I haven't yet thought through how best to handle the zone/assessment question duplication with the DB -- separate tables, a flag in the existing tables, etc? It's also not clear to me whether we should also be duplicating some/all of the information in the assessments table as well.

File: apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts (hunk 2 above)
Comment (mwest1066) [reply apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts:54]: That sounds plausible to me, but I'd appreciate @jonatanschroeder's thoughts on this too.

File: apps/prairielearn/src/components/QuestionScore.html.ts [hunk 3]
```diff
@@ -249,12 +249,33 @@ export function ExamQuestionScore({
...
+  if (
+    instance_question.status === 'saved' &&
+    !assessment_question.max_auto_points &&
+    assessment_question.max_manual_points
+  ) {
+    return html`
+      <span class="align-middle">
+        <span class="badge badge-success">saved for manual grading</span>
```
Comment (mwest1066) [inline apps/prairielearn/src/components/QuestionScore.html.ts:274]: This caused overflow for me:
image

**Review by mwest1066 — APPROVED** [review]
It worked as expected for me in testing and the code looks sensible. I'll also tag @jonatanschroeder for a second review.

File: apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts (hunk 2 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts:54]: Indeed the proper check based on points would be:
```suggestion
      } else if (question.max_auto_points || !question.max_manual_points) {
```
I believe if the status is 'saved', the question is presumably implied to be open. So a check for open may not be necessary in this context.

Comment (nwalters512) [reply apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts]: Changed in 8acfdbf.

File: apps/prairielearn/src/components/QuestionScore.html.ts (hunk 3 above)
Comment (nwalters512) [reply apps/prairielearn/src/components/QuestionScore.html.ts:233]: What do you think about changing "Submission status" to just "Status"? If we did this, I'd make the same change on the assessment instance page too.

Comment (nwalters512) [reply apps/prairielearn/src/components/QuestionScore.html.ts:233]: Discussed with @mwest1066, we'll make this change.

Comment (nwalters512) [reply apps/prairielearn/src/components/QuestionScore.html.ts:233]: I opened https://github.com/PrairieLearn/PrairieLearn/pull/10831 to make this change.

Comment (mwest1066) [reply apps/prairielearn/src/components/QuestionScore.html.ts:233]: 🎉

**Pull request merged**